sdc_client/async_api_client.py

```diff
@@ -0,0 +1,141 @@
+import asyncio
+import aiohttp
+import json
+
+from sdc_client.base_api_client import _BaseStreamSetsApiClient, UnauthorizedException, ApiClientException
+from sdc_client.interfaces import IStreamSets
+
+
+MAX_TRIES = 3
+
+
+def async_endpoint(func):
+    """
+    logs errors and returns json response
+    """
+    async def wrap(*args, **kwargs):
+        async with args[0].session:
+            for i in range(MAX_TRIES):
+                try:
+                    res = await func(*args, **kwargs)
+                    res.raise_for_status()
+                    if res.text:
+                        return await res.json()
+                except aiohttp.ClientConnectionError:
+                    if i == MAX_TRIES - 1:
+                        raise
+                    asyncio.sleep(2 ** i)
+                    continue
+                except aiohttp.ClientResponseError:
+                    if res.text:
+                        await _parse_aiohttp_response_errors(res)
```

Comment:
Just a question. If there is no return at this point, is it necessary to await parsing response?

Reply:
Yes, that is because it is declared as async. 
But if you remove **async** in a declaration and **await** in await result.json() - then **_parse_aiohttp_response_errors()** and **_parse_error_response()** become identical functions

So, my point was if it is not necessary asynchronously parse the response, we can move parsing into a base class